ravedude/src/board.rs

```diff
@@ -1,439 +1,89 @@
-}
-
-// ----------------------------------------------------------------------------
-
-struct ArduinoUno;
-
-impl Board for ArduinoUno {
-    fn display_name(&self) -> &str {
-        "Arduino Uno"
-    }
-
-    fn needs_reset(&self) -> Option<&str> {
-        None
-    }
-
-    fn avrdude_options(&self) -> avrdude::AvrdudeOptions {
-        avrdude::AvrdudeOptions {
-            programmer: "arduino",
-            partno: "atmega328p",
-            baudrate: None,
-            do_chip_erase: true,
-        }
-    }
-
-    fn guess_port(&self) -> Option<anyhow::Result<std::path::PathBuf>> {
-        Some(find_port_from_vid_pid_list(&[
-            (0x2341, 0x0043),
-            (0x2341, 0x0001),
-            (0x2A03, 0x0043),
-            (0x2341, 0x0243),
-        ]))
-    }
-}
-
-struct ArduinoMicro;
-
-impl Board for ArduinoMicro {
-    fn display_name(&self) -> &str {
-        "Arduino Micro"
-    }
-
-    fn needs_reset(&self) -> Option<&str> {
-        Some("Reset the board by pressing the reset button once.")
-    }
-
-    fn avrdude_options(&self) -> avrdude::AvrdudeOptions {
-        avrdude::AvrdudeOptions {
-            programmer: "avr109",
-            partno: "atmega32u4",
-            baudrate: Some(115200),
-            do_chip_erase: true,
-        }
-    }
-
-    fn guess_port(&self) -> Option<anyhow::Result<std::path::PathBuf>> {
-        Some(find_port_from_vid_pid_list(&[
-            (0x2341, 0x0037),
-            (0x2341, 0x8037),
-            (0x2A03, 0x0037),
-            (0x2A03, 0x8037),
-            (0x2341, 0x0237),
-            (0x2341, 0x8237),
-        ]))
-    }
-}
-
-struct ArduinoNano;
-
-impl Board for ArduinoNano {
-    fn display_name(&self) -> &str {
-        "Arduino Nano"
-    }
-
-    fn needs_reset(&self) -> Option<&str> {
-        None
-    }
-
-    fn avrdude_options(&self) -> avrdude::AvrdudeOptions {
-        avrdude::AvrdudeOptions {
-            programmer: "arduino",
-            partno: "atmega328p",
-            baudrate: Some(57600),
-            do_chip_erase: true,
+use std::collections::HashMap;
+
+use crate::config;
+
+fn get_all_boards() -> anyhow::Result<HashMap<String, config::BoardConfig>> {
+    toml::from_str(include_str!("boards.toml")).map_err(|err| {
+        if cfg!(test) {
+            anyhow::anyhow!(
+                "boards.toml in ravedude source is invalid.\n{}",
+                err.message()
+            )
+        } else {
+            anyhow::anyhow!(
+                "boards.toml in ravedude source is invalid. This is a bug, please report it!\n{}",
+                err.message()
+            )
         }
-    }
-
-    fn guess_port(&self) -> Option<anyhow::Result<std::path::PathBuf>> {
-        Some(Err(anyhow::anyhow!("Not able to guess port")))
-    }
+    })
 }
 
-struct ArduinoNanoNew;
+pub fn get_board(board_name: Option<&str>) -> anyhow::Result<config::RavedudeConfig> {
+    Ok(match board_name {
+        Some(board_name) => {
```

Comment:
This match arm is just for the legacy usecase, right?

Wouldn't it make more sense to split this into separate functions?  The newer code from `None =>` also feels to me like it should be a part of `config.rs` instead.

Reply:
Done :+1:. Yes, the match arm was for passing a named board. Now there's two functions, `get_board_from_name` and `get_board_from_manifest`. They're still both in `board.rs` because I feel like `config.rs` should be relatively separate from how it's actually read from the filesystem (`get_all_boards` which parses the included `boards.toml` is also in `board.rs`).